jwst/datamodels/tests/test_schema.py

```diff
@@ -1,73 +1,39 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
-
 from datetime import datetime
 import os
-import shutil
-import tempfile
-import warnings
 
-import asdf
 from asdf import schema as mschema
-import pytest
-import numpy as np
-from numpy.testing import assert_array_equal, assert_array_almost_equal
-import jsonschema
-from astropy.io import fits
-from astropy.modeling import models
 from astropy import time
+from astropy.io import fits
+from numpy.testing import assert_array_equal, assert_array_almost_equal
 from stdatamodels import validate
-from stdatamodels.util import gentle_asarray
-from stdatamodels.schema import merge_property_trees, build_docstring
-from stdatamodels import DataModel
+import jsonschema
+import numpy as np
+import pytest
 
 from jwst.datamodels import util
 from jwst.datamodels import _defined_models as defined_models
-from jwst.datamodels import (JwstDataModel, ImageModel, RampModel, MaskModel, MultiSlitModel,
-    AsnModel, CollimatorModel, SourceModelContainer, MultiExposureModel,
+from jwst.datamodels import (JwstDataModel, ImageModel, MaskModel, MultiSlitModel,
+    AsnModel, SourceModelContainer, MultiExposureModel,
     DrizProductModel, MultiProductModel, MIRIRampModel)
 
 
-FITS_FILE = None
-MASK_FILE = None
-TMP_FITS = None
-TMP_FITS2 = None
-TMP_YAML = None
-TMP_ASDF = None
-TMP_DIR = None
-
-def setup():
-    global FITS_FILE, MASK_FILE, TMP_DIR, TMP_FITS, TMP_YAML, TMP_ASDF, TMP_FITS2
-    ROOT_DIR = os.path.join(os.path.dirname(__file__), 'data')
-    FITS_FILE = os.path.join(ROOT_DIR, 'test.fits')
-    MASK_FILE = os.path.join(ROOT_DIR, 'mask.fits')
-
-    TMP_DIR = tempfile.mkdtemp()
-    TMP_FITS = os.path.join(TMP_DIR, 'tmp.fits')
-    TMP_FITS2 = os.path.join(TMP_DIR, 'tmp2.fits')
-    TMP_YAML = os.path.join(TMP_DIR, 'tmp.yaml')
-    TMP_ASDF = os.path.join(TMP_DIR, 'tmp.asdf')
-
-
-def teardown():
-    shutil.rmtree(TMP_DIR)
-
-
-def test_choice():
-    with pytest.raises(jsonschema.ValidationError):
-        with JwstDataModel(FITS_FILE, strict_validation=True) as dm:
-            assert dm.meta.instrument.name == 'MIRI'
+def test_strict_validation_enum():
```

Comment:
Would it be reasonable to dissolve this file and divided its tests among other jwst modules and stdatamodels?  Here's a sketch of where we might put them:

**stdatamodels tests/test_validation.py**
test_strict_validation_enum
test_strict_validation_type
test_strict_validation_date

**jwst jwst/datamodels/tests/test_models.py**
test_mask_model
test_table_size_zero

**jwst, maybe a file specific to testing the asdf extension?**
test_datamodel_schema_entry_points

**jwst, maybe a test_multislit.py?**
test_multislit_model
test_copy_multslit
test_multislit_move_from_fits
test_multislit_append_string
test_all_datamodels_init

**???**
test_data_array -- not sure what this is about

**possible duplicates**
test_list and test_list2 -- we have some tests in stdatamodels for these already
test_invalid_fits -- there's good coverage of PASS_INVALID_VALUES and STRICT_VALIDATION in stdatamodels

Reply:
Excellent.  This is done.  I'm not sure what `test_data_array` does either.  Broadcasting?  It's very odd and seems to be doing lots of things.  For now I'm leaving it where it is.